### ISSUE TITLE
Error creating volume - Specified pool is in creating state, hence volume cannot be created

### ISSUE THREAD

**Issue by okrause** [NetApp/terraform-provider-netapp-gcp#77]
I am creating a regional storage pool and a volume inside it. Storage Pool creating is still in progress, but somehow the provider thinks it is done and continues with volume. Then, it fails because pool creation is still in progress.

```bash
netapp-gcp_storage_pool.regional-storage-pool: Creation complete after 5m27s [id=692608a9-ee39-d4df-c466-0b7b9a4a702c]
netapp-gcp_volume.gcp-volume-nfs: Creating...
╷
│ Error: code: 400, message: Error creating volume - Specified pool is in creating state, hence volume cannot be created
```

Indeed, the pool creation took much longer.
Reapplying the TF file after the pool was created works fine. Something is wrong with state recognition of the pool creation.

Here the TF file used:
```terraform
# This example creates a regional Storage Pool

# local variables
locals {
  network = "cvs-prd-shared-vpc"
  pool_name = "sp_regional_example"
  region = "europe-west1"
  zone = "europe-west1-b"
  secondary_zone = "europe-west1-c"
  pool_size = 1024
  volume_name = "tf-regional-volume"
  volume_size = 100
}

resource "netapp-gcp_storage_pool" "regional-storage-pool" {
  name = local.pool_name
  region = local.region
  zone = local.zone
  secondary_zone = local.secondary_zone
  network = local.network
  size = local.pool_size
  service_level = "ZoneRedundantStandardSW"
  storage_class = "software"
}

resource "netapp-gcp_volume" "gcp-volume-nfs" {
  # This block mirrors pool settings
  region = netapp-gcp_storage_pool.regional-storage-pool.region
  zone = netapp-gcp_storage_pool.regional-storage-pool.zone
  network = netapp-gcp_storage_pool.regional-storage-pool.network
  pool_id = netapp-gcp_storage_pool.regional-storage-pool.id
  regional_ha = true
  storage_class = "software" 
  service_level = "standard"  # leave as is

  # Volume specific settings
  name = local.volume_name
  protocol_types = ["NFSv3"]
  size = local.volume_size

  # Snapshot policy definition
  snapshot_policy {
    enabled = true
  }
}
```

**Comment by wenjun666** [comment]
There is wait for completion mechanism in pool resource. Is it possible that you need add depends on to volume creation? Otherwise terraform make run two tasks concurrently.

**Comment by okrause** [comment]
pool_id in resource netapp-gcp_volume is using netapp-gcp_storage_pool.regional-storage-pool.id. Thats an implicit dependency of the resources which TF resolves. So it should work without an explicit dependency.

**Comment by okrause** [comment]
I retested with tracing on and found the issue:

```bash
2022-09-23T11:57:20.846+0200 [DEBUG] provider.terraform-provider-netapp-gcp_v22.8.2: 2022/09/23 11:57:20 Job is still onging, return after maximum wait time is reached.
2022-09-23T11:57:20.846+0200 [DEBUG] provider.terraform-provider-netapp-gcp_v22.8.2: 2022/09/23 11:57:20 Created storage pool in region: europe-west1
netapp-gcp_storage_pool.regional-storage-pool: Creation complete after 5m23s [id=ce1d1d1c-81eb-1284-b0d5-d7fb59109ff6]
```

https://github.com/NetApp/terraform-provider-netapp-gcp/blob/d6a3783eee272305e90a93fb4cc04a7a47e1bbbb/gcp/storage_pool.go#L63
sets a 300s timeout. That's too short for storage pools. My recommendation is to set a 20min limit.

Looking at https://github.com/NetApp/terraform-provider-netapp-gcp/blob/d6a3783eee272305e90a93fb4cc04a7a47e1bbbb/gcp/volumeReplication.go#L148, I see two issues:
1. Type in line 166 "onging" versus "ongoing"
2. line 167: We returning nil (=all good) if timeout expires. To me, a timeout is an error and it should stop execution.

**Comment by okrause** [comment]
Fixed in v22.10.0. Closing.
